src/stash/src/postgres.rs

```diff
@@ -502,16 +538,18 @@ impl Postgres {
         let client = self.client.as_mut().unwrap();
         let stmts = self.statements.as_ref().unwrap();
         let stmts = CountedStatements::from(stmts);
-        let tx = client
-            .build_transaction()
-            .read_only(self.readonly)
-            .start()
-            .await?;
+        // Generate statements to execute depending on our mode.
+        let (tx_start, tx_end) = match self.txn_mode {
+            TransactionMode::Writeable => ("BEGIN", "COMMIT"),
+            TransactionMode::Readonly => ("BEGIN READ  ONLY", "ROLLBACK"),
```

Comment:
This is a good point! I had thought that it was double protection against accidentally committing, but COMMIT is needed to ensure all the previous statements were executed correctly, in case cockroach needed to bump this txn's timestamp or something. Changed.